src/webgl/ShaderGenerator.js

```diff
@@ -1638,3 +1638,509 @@ export default shadergenerator;
+ * - `canvasContent`: a texture containing the sketch's contents before the filter is applied.
+ *
+ * Return a four-component vector `[r, g, b, a]` for the pixel.
+ *
+ * This hook is available in:
+ * - <a href="#/p5/baseFilterShader">baseFilterShader()</a>
+ *
+ * @param {Function} callback
+ *        A callback function which receives the inputs object and canvasContent, and should return a color array.
+ *
+ * @example
+ * <div modernizr='webgl'>
+ * <code>
+ * let myShader;
+ * function setup() {
+ *   createCanvas(200, 200, WEBGL);
+ *   myShader = baseFilterShader().modify(() => {
+ *     getColor((inputs, canvasContent) => {
+ *       // Warp the texture coordinates for a wavy effect
+ *       let warped = [inputs.texCoord.x, inputs.texCoord.y + 0.1 * sin(inputs.texCoord.x * 10.0)];
+ *       return getTexture(canvasContent, warped);
+ *     });
+ *   });
+ * }
+ * function draw() {
+ *   background(180);
+ *   // Draw something to the canvas
+ *   fill('yellow');
+ *   circle(0, 0, 150);
+ *   filter(myShader);
+ * }
+ * </code>
+ * </div>
+ */
+
+/**
+ * @method getObjectInputs
+ * @description
+ * Registers a callback to modify the properties of each vertex before any transformations are applied in the vertex shader. This hook can be used inside <a href="#/p5/baseColorShader">baseColorShader()</a>.modify() and similar shader <a href="#/p5.Shader/modify">modify()</a> calls to move, color, or otherwise modify the raw model data. The callback receives an object with the following properties:
+ * 
+ * - `position`: a three-component vector representing the original position of the vertex.
+ * - `normal`: a three-component vector representing the direction the surface is facing.
+ * - `texCoord`: a two-component vector representing the texture coordinates.
+ * - `color`: a four-component vector representing the color of the vertex (red, green, blue, alpha).
+ *
+ * Return the modified object to update the vertex.
+ *
+ * This hook is available in:
+ * - <a href="#/p5/baseColorShader">baseColorShader()</a>
+ * - <a href="#/p5/baseMaterialShader">baseMaterialShader()</a>
+ * - <a href="#/p5/baseNormalShader">baseNormalShader()</a>
+ * - <a href="#/p5/baseStrokeShader">baseStrokeShader()</a>
+ *
+ * @param {Function} callback
+ *        A callback function which receives the vertex object and should return it after making any changes.
+ *
+ * @example
+ * <div modernizr='webgl'>
+ * <code>
+ * let myShader;
+ * function setup() {
+ *   createCanvas(200, 200, WEBGL);
+ *   myShader = baseColorShader().modify(() => {
+ *     let t = uniformFloat(() => millis());
+ *     getObjectInputs(inputs => {
+ *       // Create a sine wave along the x axis in object space
+ *       inputs.position.y += 3 * sin(t * 0.001 + inputs.position.x * 0.05);
+ *       return inputs;
+ *     });
+ *   });
+ * }
+ * function draw() {
+ *   background(220);
+ *   shader(myShader);
+ *   noStroke();
+ *   fill('orange');
+ *   sphere(100);
```

Comment:
@perminder-17 mentioned it on another example, but on here too `sphere(100)` fills the canvas entirely, I think `sphere(50)` leaves some room for it to go up and down:

https://github.com/user-attachments/assets/ee838bc6-eceb-444f-9a0d-2208cc7bb3af